MF-448 - Add option to connect to DB with verify-ca and verify-full

mfinley3 opened the pull request:
### What does this do?
Adds the option to connect to the DBs with verify-ca and verify-full
Users can now specify any extra certs and keys they may need.

### Which issue(s) does this PR fix/relate to?
Resolves #448 

### List any changes that modify/break current functionality
None of the default behavior has changed, the default db ssl connection mode remains `disabled`

### Have you included tests for your changes?
I have updated the relevant tests. The pq library tests should cover the changes here since these are just implementing their options

### Did you document any new/modified functionality?
Yes!

### Notes
No other notes

mfinley3 committed:
MF-448 - Add option to connect to DB with verify-ca and verify-full

Adds the option to connect with verify-ca and verify-full
Users can now specify any extra certs and keys they may need.

Signed-off-by: MichaelFinley <Michael.Finley@target.com>

dborovcanin reviewed (dismissed):
Have you considered wrapping database connection parameters to some kind of config entity (map, for example)?

anovakovic01 commented on users/postgres/init.go:
```diff
@@ -18,8 +18,8 @@ import (
 // failure.
-func Connect(host, port, name, user, pass, sslMode string) (*sql.DB, error) {
-	url := fmt.Sprintf("host=%s port=%s user=%s dbname=%s password=%s sslmode=%s", host, port, user, name, pass, sslMode)
+func Connect(host, port, name, user, pass, sslMode, sslCert, sslKey, sslRootCert string) (*sql.DB, error) {
```
I think that it makes sense to create db config structure and use it as parameter here.

chombium replied:
@anovakovic01 what do you mean by "db config structure"? I hope you are not thinking of storing the config in some db table ;)
If we need config lets do it as @srados suggested in #435

anovakovic01 replied:
@chombium yes, I meant passing struct instead of all these parameters.

chombium replied:
@chombium, @srados Thanks for the clarification. I thought that too, but that db caught my eyes :)

mfinley3 committed:
Passing db config struct rather than seperate parameters

Also updated the tests to use the config

Signed-off-by: MichaelFinley <Michael.Finley@target.com>

xuantinfx reviewed (dismissed):
ok

anovakovic01 commented on cmd/users/main.go:
```diff
@@ -34,40 +34,41 @@ import (
+	dbConfig   postgres.Config
 	HTTPPort   string
 	GRPCPort   string
 	Secret     string
```
Please, change other `config` fields to private.

mfinley3 replied:
Done!

anovakovic01 commented on things/postgres/init.go:
```diff
@@ -15,11 +15,23 @@ import (
 	migrate "github.com/rubenv/sql-migrate"
 )
 
+type Config struct {
```
Add comment to explain meaning of this config struct.

mfinley3 replied:
Added a comment to both this one and users/postgres/init.go

anovakovic01 reviewed (changes requested):
@mfinley3 Almost there.

mfinley3 committed:
Unexporting fields apart of users config

Also added comments to the newly exported Config in things & users postgres/init.go

Signed-off-by: MichaelFinley <Michael.Finley@target.com>

drasko reviewed (approved):
LGTM

anovakovic01 reviewed (approved):
LGTM!
